tests/neg-macros/i11483/Test_2.scala

```diff
@@ -0,0 +1,28 @@
+package x
+
+import scala.language.implicitConversions
+import scala.concurrent.Future
+
+given FutureAsyncMonad: CpsMonad[Future] with
+  def pure[T](t:T): Future[T] = ???
+  def impure[T](t:Future[T]): T = ???
+  def map[A,B](x:Future[A])(f: A=>B): Future[B] = ???
+
+
+object Api:
+
+  def doSomething():Future[String] =
+    Future.successful("doSomething")
+
+  def println(x:String):Unit =
+    Console.println(x)
+
+
+object Main:
+
+  def main(args: Array[String]): Unit =
+    X.process[Future,Unit]{
+      Api.println( // error
+         Api.doSomething() // error
+      )
```

Comment:
It's not obvious why `x.Api.println` needs eta-expansion. Is it due to bugs in the macros?

Reply:
That is because it is passed as a function to some other method. It is a bug in the macro as I explained in the issue and there is a simple solution to fix the macro.